src/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanner.java

```diff
@@ -46,8 +47,22 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
 
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		boolean includeErrorResponses=true;
+		switch (this.getLevel()) {
+			case HIGH:	includeErrorResponses=false; break;  
+			case MEDIUM: 					
+			case DEFAULT: 
+			case LOW: 		
+			case OFF: } 
 		if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isText()){
-			Vector<String> xFrameOption = msg.getResponseHeader().getHeaders(HttpHeader.X_FRAME_OPTION);
+			int responseStatus = msg.getResponseHeader().getStatusCode();
+			// If it's an error and we're not including error responses then just return without alerting
+			if ((HttpStatusCode.isServerError(responseStatus) ||
+					HttpStatusCode.isClientError(responseStatus)) &&
+					!includeErrorResponses) {

```

Comment:
Consider switching the operands of operator ```&&```, would avoid checking the status when the errors are included (which is probably faster).